Sources/SwiftProtobuf/JSONScanner.swift

```diff
@@ -220,73 +222,81 @@ private func decodeString(_ s: String) -> String? {
 // If it encounters a "\" backslash character, it returns a nil.  This
 // is used by callers that are parsing quoted numbers.  See nextSInt()
 // and nextUInt() below.
-private func parseBareUInt(p: inout UnsafePointer<UInt8>, end: UnsafePointer<UInt8>) throws -> UInt64? {
-    let start = p
-    let c = p[0]
-    p = p + 1
-    switch c {
-    case asciiZero: // 0
-        if p != end {
-            let after = p[0]
-            switch after {
-            case asciiZero...asciiNine: // 0...9
-                // leading '0' forbidden unless it is the only digit
-                throw JSONDecodingError.leadingZero
-            case asciiPeriod, asciiLowerE: // . e
-                // Slow path: JSON numbers can be written in floating-point notation
-                p = start
-                if let s = try parseBareFloatString(p: &p, end: end) {
-                    if let d = Double(s) {
-                        if let u = UInt64(exactly: d) {
-                            return u
-                        }
-                    }
-                }
-                throw JSONDecodingError.malformedNumber
-            case asciiBackslash:
-                return nil
-            default:
-                return 0
+private func parseBareUInt(
+  source: UnsafeBufferPointer<UInt8>,
+  index: inout UnsafeBufferPointer<UInt8>.Index,
+  end: UnsafeBufferPointer<UInt8>.Index
```

Comment:
I thought `parseBareUInt` was always called with `end` referring to the end of the input.  If I'm right, then `end` here may be redundant, you can just use `source.endIndex` within this function and lose the extra argument.